### PR TITLE
fix(config): prevent vite from loading config

### LINKED ISSUE #10262: Astro does not respect vite `build`/`serve` commands
### Astro Info

```block
Astro                    v4.3.6
Node                     v20.11.1
System                   macOS (x64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             none
```

### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

When astro loads the vite config it does not set the correct `command` from vite's `UserConfig`.

For instance, running `astro build` with a plugin that has `apply: "serve"` will load the plugin nonetheless.

```js
// vite.config.js
definedConfig({
  plugins: [
    { apply: "serve",  ...somePlugin() }
 ]
});
```

will apply `somePlugin`. Same goes with `command` is read and used instead of `apply`.

### What's the expected result?

Astro should set the correct `command` when loading the vite config.

NOTE: I could not add a minimal reproducible example because the playground seems broken (errors out on the default example):

```
~/projects/github-ngr3mk
❯ astro build
ech[callSiteLocation@
depd@
@
@https://githubngr3mk-ie4y.w-corp-staticblitz.com/blitz.6f234770.js:54:14872
@https://githubngr3mk-ie4y.w-corp-staticblitz.com/blitz.6f234770.js:54:15558
@https://githubngr3mk-ie4y.w-corp-staticblitz.com/blitz.6f234770.js:54:13457
@https://githubngr3mk-ie4y.w-corp-staticblitz.com/blitz.6f234770.js:54:10539
@https://githubngr3mk-ie4y.w-corp-staticblitz.com/blitz.6f234770.js:54:13780
@
@https://githubngr3mk-ie4y.w-corp-staticblitz.com/blitz.6f234770.js:54:14872
@https://githubngr3mk-ie4y.w-corp-staticblitz.com/blitz.6f234770.js:54:15558
@https://githubngr3mk-ie4y.w-corp-staticblitz.com/blitz.6f234770.js:54:13457
@https://githubngr3mk-ie4y.w-corp-staticblitz.com/blitz.6f234770.js:54:10539
@https://githubngr3mk-ie4y.w-corp-staticblitz.com/blitz.6f234770.js:189:2456
@https://githubngr3mk-ie4y.w-corp-staticblitz.com/blitz.6f234770.js:352:378392
@https://githubngr3mk-ie4y.w-corp-staticblitz.com/blitz.6f234770.js:352:379993
@https://githubngr3mk-ie4y.w-corp-staticblitz.com/blitz.6f234770.js:352:378526]
o
~/projects/github-ngr3mk 1s
❯ echo $?
1
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-ngr3mk

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by nmattia**
Fixes #10262

## Changes

This fixes an issue where creating a vite server (to e.g. load a TypeScript astro config) would trigger vite to look up and load an unrelated vite config (potentially going up directories).

## Testing

This has not been tested because I'm not familiar with the astro testing setup...

## Docs

This fixes an unexpected side effect of loading a TypeScript config. Since this is the expected behavior, it might not require documentation.

### CODE REVIEW

File: .changeset/khaki-emus-wait.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"astro": patch
+---
+
+Instruct vite to not look for an external vite config file when creating a server
```
Comment (lilnasy) [inline .changeset/khaki-emus-wait.md:5]:
```suggestion
Fixes an issue where some CLI commands attempted to directly read vite config files.
```

**Review by lilnasy — APPROVED** [review]
Looks great to me! Thanks for taking this on.

File: .changeset/khaki-emus-wait.md (hunk 1 above)
Comment (lilnasy) [reply .changeset/khaki-emus-wait.md:5]: Trying to rephrase it for users rather than maintainers.

**Comment by nmattia** [comment]
@lilnasy thanks, PTAL! as far as I'm concerned you can merge

File: packages/astro/src/core/config/vite-load.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@ import { debug } from '../logger/core.js';
 
 async function createViteServer(root: string, fs: typeof fsType): Promise<ViteDevServer> {
 	const viteServer = await createServer({
+		configFile: false /* prevent vite from trying to discover a vite conf file */,
```
Comment (bluwy) [inline packages/astro/src/core/config/vite-load.ts:9]:
```suggestion
		configFile: false,
```
I don't think we need an explanation here, I think it's clear what the intent is.

**Pull request merged**